internal/elasticsearch/service.go

```diff
@@ -0,0 +1,112 @@
+package elasticsearch
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ViaQ/logerr/kverrors"
+	"github.com/ViaQ/logerr/log"
+	"github.com/openshift/elasticsearch-operator/internal/manifests/service"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// CreateOrUpdateServices ensures the existence of the services for Elasticsearch cluster
+func (er *ElasticsearchRequest) CreateOrUpdateServices() error {
+	dpl := er.cluster
+
+	annotations := make(map[string]string)
+	serviceName := fmt.Sprintf("%s-%s", dpl.Name, "cluster")
+
+	errCtx := kverrors.NewContext("service_name", serviceName,
+		"cluster", er.cluster.Name,
+		"namespace", er.cluster.Namespace,
+	)
+
+	err := er.createOrUpdateService(
```

Comment:
nit, but i don't think there's a reason to abstract this further out... we should be making use of the generic service CreateOrUpdate function within here

Reply:
It is actually using the generic one below.

Reply:
right, my point was more that there wasn't a huge need to use `es.createOrUpdateService` and that the logic contained in there could have stayed in this function